intake_cesm/manage_collections.py

```diff
@@ -0,0 +1,301 @@
+import fnmatch
+import logging
+import os
+import shutil
+import re
+
+from subprocess import Popen, PIPE
+
+import pandas as pd
+import yaml
+from tqdm import tqdm
+
+logging.basicConfig(level=logging.DEBUG)
+
+class StorageResource(object):
+    def __init__(self, urlpath, type, file_extension=".nc"):
+        self.urlpath = urlpath
+        self.type = type
+        self.file_extension = file_extension
+
+        self.filelist = self._list_files()
+
+    def _list_files(self):
+        if self.type == "posix":
+            return self._list_files_posix()
+
+        if self.type == "hsi":
+            return self._list_files_hsi()
+
+    def _list_files_posix(self):
+        """Get a list of files"""
+        w = os.walk(self.urlpath)
+
+        filelist = []
+        for root, dirs, files in w:
+            filelist.extend([
+                os.path.join(root, f)
+                for f in files if f.endswith(self.file_extension)
+                ]
+                            )
+
+        return filelist
+
+
+    def _list_files_hsi(self):
+
+        if shutil.which("hsi") is None:
+            logging.warning(f"no hsi; cannot access [HSI]{self.urlpath}")
+            return []
+
+        p = Popen([
+            "hsi",
+            'find {urlpath} -name "*{file_extension}"'.format(
+                urlpath=self.urlpath,
+                file_extension=self.file_extension)
+            ], stdout=PIPE, stderr=PIPE)
+
+        stdout, stderr = p.communicate()
+        lines = stderr.decode("UTF-8").strip().split("\n")[1:]
+
+        filelist = []
+        i = 0
+        while i < len(lines):
+            if "***" in lines[i]:
+                i += 2
+                continue
+            else:
+                filelist.append(lines[i])
+                i +=1
+
+        return filelist
+
+
+
+class CESMCollections(object):
+    db_dir = "./collections"
```

Comment:
Having `db_dir` as a relative path might create problems during/after package installation. My recommendation would be to use a "stable" and absolute location path such as `~/.intake-cesm/collections`

Reply:
I would like take the first point in my previous comment back :)

> 1. We can get rid of `config.py` since we were doing very minimal work in that module. As you can see we are just returning a dictionary in `config.py`. 

We can keep `config.py` and use it for some package configurations such as creating the `~/.intake-cesm/collections` I mentioned above. 

```python
import os

HOME = os.environ.get("HOME", None)
db_dir = f"{HOME}/.intake-cesm/collections"
os.makedirs(db_dir, exists_ok=True)
```

With this change implemented, we could just import `db_dir` in `manage_collections.py`:
```python
from .config import db_dir
```

Let me know what you think.

Reply:
I have implemented a `SETTINGS` directory in `config.py`. This is settable via a `set_options` class, inspired by `xarray`. 

Test collection are now stored in tests/test_collections. The database_directory is set in tests/__init__.py as follows
```
set_options(database_directory='./test_collections')
```

This seems to be a good solution.